src/gwt/src/org/rstudio/core/client/files/FileSystemItem.java

```diff
@@ -132,6 +136,17 @@ public final FileSystemItem getParentPath()
       }
    }
 
+   public final String getRootPath()
```

Comment:
Isn't this the same as `getName()`? (except that it doesn't handle paths like `/foo/bar/`)

I also don't think this is the right name for such a method; the "root" of a path is usually either `/` or `C:/` or `//server/blah`.

Reply:
You're right this is a bad name -- I'm open to suggestions here for better names. I kind of felt the name wasn't right that's why Ieft the comment here. But it's not the same as getName() in many cases. 

The idea is we need to handle paths like `foo/bar/baz`, and need the inverse of `getParentPath()` so `getParentPath('foo/bar/baz')` will return `foo/bar`, we want a method that returns `baz`, basically the last part of the path (either the immediate dir name if a dir or the file name if a file). 

Needed this because basically, when creating the Rproj file, we took the dir name: `foo/bar/baz` and added `.Rproj` to make it `foo/bar/baz.Rproj` and then combined with the directory name which resulted in paths like `foo/bar/baz/foo/bar/baz.Rproj` instead of `foo/bar/baz/baz.Rproj`. This didn't seem right, so just needed a method to pull out the last part of the path if the absolute path is passed in.

Reply:
Actually now realizing that getName which calls getNameFromPath was probably failing and returning the full path only because of Windows using the wrong separators `\` instead of `/` and with fixing the separator issue with `normalizeSeparators` it probably should have worked correctly, but I fixed the path issue before fixing the issue of the separators. I'll clean this up because makes sense now -- getName should probably have worked if the path was normalized, but with Windows style paths it was returning the full absolute path resulting in our yucky duplicated path issues

Reply:
Thanks! That all makes sense to me.